File: 1614. Maximum Nesting Depth of the Parentheses/Solution.py
class Solution(object):
    def maxDepth(self, s):
        """
        :type s: str
        :rtype: int
        """
        d = 0
        res = 0

        for i in s:
            if i=='(':
                d+=1
            if i==')':
                d-=1

            res = max(d,res)

        return res
        

File: 1614. Maximum Nesting Depth of the Parentheses/Solution.java
class Solution {
    public int maxDepth(String s) {
        if(s.length()==0)return 0;
        int max = 0;
        int depth = 0;
        for(char i : s.toCharArray()){
            if(i=='('){
                depth++;
            }

            else if(i==')'){
                depth--;
            }
            max = Math.max(max,depth);
        }

        return max;
    }
}